Client/mods/deathmatch/logic/lua/CLuaArgument.cpp

```diff
@@ -176,20 +170,19 @@ bool CLuaArgument::CompareRecursive(const CLuaArgument& Argument, std::set<CLuaA
             if (m_pTableData->Count() != Argument.m_pTableData->Count())
                 return false;
 
-            vector<CLuaArgument*>::const_iterator iter = m_pTableData->IterBegin();
-            vector<CLuaArgument*>::const_iterator iterCompare = Argument.m_pTableData->IterBegin();
-            while (iter != m_pTableData->IterEnd() && iterCompare != Argument.m_pTableData->IterEnd())
-            {
-                if (pKnownTables->find(m_pTableData) == pKnownTables->end())
-                {
-                    pKnownTables->insert(m_pTableData);
-                    if (*iter != *iterCompare)
-                        return false;
-                }
+            auto iter = m_pTableData->begin();
+            auto iterCompare = Argument.m_pTableData->begin();
+            if (pKnownTables->find(m_pTableData) != pKnownTables->end())
+                return true;
+
+            pKnownTables->insert(m_pTableData);
```

Comment:
You're doing the double work here. std::set::insert is doing the same find as you have already done above. You can insert a value and use the second attribute of the returned value to check if the value was already inserted.

Reply:
Would `return m_pTableData == Argument.m_pTableData;` work here instead of the loop too?

> Yes, `operator==` is correctly defined for all standard containers (**except** the unordered containers - based on 23.2.5.2 of the standard), and will generally do a lexicographic comparison. See for example [here](http://en.cppreference.com/w/cpp/container/set/operator_cmp). The relevant quote:
>> Checks if the contents of lhs and rhs are equal, that is, whether lhs.size() == rhs.size() and each element in lhs has equivalent element in rhs at the same position.
> 
> Since `std::set` is an ordered container, any set with the same size and same elements (given the comparators are the same) will necessarily have them in the same position, hence will compare equal.

Reply:
Yes, it should work.